sentry_sdk/integrations/opentelemetry/span_processor.py

```diff
@@ -90,6 +96,22 @@ def global_event_processor(event, hint):
             # type: (Event, Hint) -> Event
             return link_trace_context_to_error_event(event, self.otel_span_map)
 
+    def _prune_old_spans(self):
+        # type: (SentrySpanProcessor) -> None
+        """
+        Prune spans that have been open for too long.
+        """
+        current_time_minutes = int(time() / 60)
+        for span_start_minutes in self.open_spans:
```

Comment:
I see you applied my change here... One thing I just thought of though is whether popping items from the `open_spans` during the iteration might lead to unexpected results. Let's make sure that this code still works as intended.

Reply:
yes, this is a problem. I am fixing this right now :-)

Reply:
Yeah, I see the tests are failing

In that case, you can go back to using `list(self.open_spans.keys)`, but please add a comment to explain that making the list is necessary since we modify the dictionary during iteration.